chainercv/links/model/feature_extraction_predictor.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+import warnings
+
+import chainer
+from chainer import cuda
+
+from chainercv.transforms import center_crop
+from chainercv.transforms import scale
+from chainercv.transforms import ten_crop
+
+
+class FeatureExtractionPredictor(chainer.Chain):
+
+    """Wrapper class that adds predict method to a feature extraction model.
+
+    The :meth:`predict` takes three steps to make predictions.
+
+    1. Preprocess images
+    2. Forward the preprocessed images to the network
+    3. Average features in the case when ten-crop is used.
+
+    Example:
+
+        >>> from chainercv.links import VGG16
+        >>> from chainercv.links import FeatureExtractionPredictor
+        >>> base_model = VGG16()
+        >>> model = FeatureExtractionPredictor(base_model)
+        >>> prob = model.predict([img])
+        # Predicting multiple features
+        >>> model.extractor.feature_names = ['conv5_3', 'fc7']
+        >>> conv5_3, fc7 = model.predict([img])
+
+    """
+
+    def __init__(self, extractor,
+                 size=(224, 224), scale_size=256,
```

Comment:
I found it convenient to support following two cases.

+ crop_size is not a square
+ resize to certain shape instead of scaling

Therefore, I changed `prepare` to support both an integer and a tuple. 4bbdd5a